server/actions/donation.ts

```
@@ -11,11 +11,13 @@ export async function logDonation({
   name,
   email,
   amount,
+  userId,
   nonprofitId
 }: DonationType): Promise<void> {
   await Mongo();
 
   const nonprofit = await Nonprofit.findOne({ _id: nonprofitId });
+  const user = await Donation.findOne({ _id: userId }); // TODO: Don't allow front end to pass null resulting userId
```

Technically this should be `User.findOne` right? Also right now we haven't decided if all Donations require a user ID, so could you make necessary changes to the Mongoose schema and the types in `utils.ts` to support `null` user ID or make user ID optional?

---

If a user ID is specified and there is no user with that ID, we should throw an error, just like on line 22 we throw an error if the nonprofit doesn't exist.